Chessington.GameEngine/Pieces/Queen.cs

```diff
@@ -6,11 +6,87 @@ namespace Chessington.GameEngine.Pieces
     public class Queen : Piece
     {
         public Queen(Player player)
-            : base(player) { }
+            : base(player)
+        {
+        }
+
```

Comment:
Great stuff! Modular and easy to read. Just a general thought for bishop, queen and rook, this might be a bit tricky to follow for someone who is not accustomed to the problem/project.

For example, understanding how you compute the nextRow and nextCol as a sum of the current one and distance multiplied with the direction.

But nonetheless, cool implementation!

Reply:
Done! The Direction enum should make it clear